__tests__/index.test.ts

```diff
@@ -232,4 +238,54 @@ describe('Test main action', () => {
         multilineInputSpy.mockClear();
         nameTransformationSpy.mockClear();
     });
+
+    
+    test('handles invalid timeout string and falls back to default', async () => {
+        const timeoutSpy = jest.spyOn(core, 'getInput').mockReturnValue(INVALID_TIMEOUT_STRING);
+
+        smMockClient
+        .on(GetSecretValueCommand)
+        .resolves({ SecretString: 'test' });
+        
+        await run();
+        
+        expect(core.getInput).toHaveBeenCalledWith('auto-select-family-attempt-timeout');
+
+        
+        timeoutSpy.mockClear();
+    });
+
+    test('handles valid timeout value', async () => {
+        const timeoutSpy = jest.spyOn(core, 'getInput').mockReturnValue(VALID_TIMEOUT);
+
+        smMockClient
+        .on(GetSecretValueCommand)
+        .resolves({ SecretString: 'test' });
+
+        await run();
+        
+        expect(core.getInput).toHaveBeenCalledWith('auto-select-family-attempt-timeout');
+
+        
+        timeoutSpy.mockClear();
+    });
+
+    test('handles invalid negative timeout value and falls back to default', async () => {
+        const timeoutSpy = jest.spyOn(core, 'getInput').mockReturnValue(INVALID_TIMEOUT_NUMBER);
+        
+        // Set up a simple mock response
+        smMockClient
+            .on(GetSecretValueCommand)
+            .resolves({ SecretString: 'test' });
+    
+        // If the request goes through, it means the timeout was successfully set to default
+        await run();
+    
+        expect(core.getInput).toHaveBeenCalledWith('auto-select-family-attempt-timeout');
```

Comment:
The expect (in here and other tests) has to be done against `setDefaultAutoSelectFamilyAttemptTimeout` not the `getInput`.